crates/project-model/src/target_data_layout.rs

```diff
@@ -32,7 +32,16 @@ pub fn get(
             Sysroot::set_rustup_toolchain_env(&mut cmd, sysroot);
             cmd.envs(extra_env);
             cmd.current_dir(cargo_toml.parent())
-                .args(["rustc", "--", "-Z", "unstable-options", "--print", "target-spec-json"])
+                .args([
+                    "rustc",
+                    "-Z",
+                    "unstable-options",
+                    "--print",
+                    "target-spec-json",
+                    "--",
+                    "-Z",
+                    "unstable-options",
+                ])
```

Comment:
Is this an accidental change (from debugging perhaps)?

Reply:
While it looks wonky, it's intended. I explained it in the [commit message](https://github.com/rust-lang/rust-analyzer/pull/16621/commits/2826eb51aa60e2e73d7f35408d2ee2727ec36334) for this change, which I'm pasting here:

> `cargo rustc -- <args>` first builds dependencies then calls `rustc <args>` for the current package. Here, we don't want to build dependencies, we just want to call `rustc --print`. An unstable `cargo rustc` `--print` command bypasses building dependencies first. This speeds up execution of this code path and ensures RA doesn't recompile dependencies with the `RUSTC_BOOTSRAP=1` env var flag set.
> 
> Note that we must pass `-Z unstable-options` twice, first to enable the `cargo` unstable `--print` flag, then later to enable the unstable `rustc` `target-spec-json` print request.

Reply:
Oh I see, I was very confused by the invoked command as it looks somewhat funny :sweat_smile: 

This is a great change thanks, I've been wondering if we could do better as one other downside of the current command is that it doesnt work in virtual workspaces, whereas this does!